web/src/lib/domain-logic/getNavBarBusinessTitle.test.tsx

```diff
@@ -1,141 +1,344 @@
 import { getMergedConfig } from "@/contexts/configContext";
 import { getNavBarBusinessTitle } from "@/lib/domain-logic/getNavBarBusinessTitle";
 import {
+  BusinessPersona,
   generateBusiness,
   LookupIndustryById,
   LookupLegalStructureById,
+  ProfileData,
 } from "@businessnjgovnavigator/shared";
 import { generateProfileData } from "@businessnjgovnavigator/shared/test";
 
 const Config = getMergedConfig();
 
+const name = "some cool name";
+
+const publicFilingProfile = (): Partial<ProfileData> => {
+  return {
+    tradeName: "",
+    legalStructureId: "limited-liability-company",
+  };
+};
+
+const tradeNameProfile = (): Partial<ProfileData> => {
+  return {
+    businessName: "",
+    legalStructureId: "sole-proprietorship",
+  };
+};
+
 describe("getNavBarBusinessTitle", () => {
-  it("shows guest text if industryId is undefined", () => {
-    const business = generateBusiness({
-      profileData: generateProfileData({
-        businessName: "",
-        tradeName: "",
-        industryId: undefined,
-      }),
-    });
-    const navBarBusinessTitle = getNavBarBusinessTitle(business);
-    expect(navBarBusinessTitle).toEqual(Config.navigationDefaults.navBarGuestText);
-  });
+  describe("when name is defined", () => {
+    describe("when legal structure undefined", () => {
+      it.each(["STARTING", "OWNING"])("shows business name", (businessPersona) => {
```

Comment:
This is a cool pattern, just to confirm it's an alternative to a loop right.

Reply:
yeah @adidner did a demo on `it.each` a while back, I think it's all over our codebase now

Reply:
oh yeah, when he first joined the team. This is the first time I'm seeing it but yeah it is throughout. ty for the clarification.